docs/tex/mcmc.tex

```diff
@@ -0,0 +1,208 @@
+Let $t \in \mathbb{R}^K$ denote the vector of values of continuous random choices in the previous model trace.
+Let $u \in \mathbb{R}^L$ denote the vector of values of continuous random choices in the proposal trace.
+Similarly, let $t' \in \mathbb{R}^{K'}$ and $u' \in \mathbb{R}^{L'}$ denote the vector of values of continuous random choices in the new model trace, and new proposal trace.
+We assume that $K + L = K' + L'$ (this is the \emph{dimension matching} requirement), and that we are given a bijection $h : \mathbb{R}^{K + L} \to \mathbb{R}^{K + L}$ that is differentiable and whose Jacobian has nonzero determinant everwhere.\footnote{It may be possible to generalize and allow functions that are not differentiable at a measure zero set of points.}
+This section discusses how to efficiently compute the absolute value of the determinant of the Jacobian of $h$, which we denote by $J \in \mathbb{R}^{K + L \times K + L}$.
+
+$J$ is often sparse due to the fact that often a large fraction of the entries in $(t', u')$ are obtained by \emph{copying} the value of some entry in $(t, u)$.
+We denote the subvector of entries of $t'$ that are produced by copying as $t'_c$ and similarly for $u'$ (subvector of copies is denoted $u'_c$).
+We denote the subvectors containing other entries of $t'$ and $u'$ by $t'_w$ and $u'_w$ respectively (where $w$ stands for `write').
+Let $t_c$ and $u_c$ denote the subvectors of $t$ and $u$ respectively such that each entry is copied to at least one entry in $(t', u')$.
+We denote the subvectors containing the remaining entries of $t$ and $u$ as $t_r$ and $u_r$ (where $r$ stands for `read').
+The Jacobian of $h$ (up to reordering of rows and columns, which does not change the absolute value of the determinant) then has the following block structure, where white space indicates a zero submatrix (and where sets of rows and columns are labeled on right):
+\[
+J = \left[
+\begin{array}{cccc}
+J_{11}  &   J_{12}  &     &   \\
+J_{21}  &   J_{22}  &     &   \\
+J_{31}  &   J_{32}  &   J_{33}  &   J_{34}\\
+J_{41}  &   J_{42}  &   J_{43} &   J_{44}
+\end{array}
+\right]
+\;\;\;\;\;
+\begin{array}{c|cccc}
+&       t_c    &   u_c &   t_r &   u_r\\
+\hline
+t'_c  &  J_{11}  &   J_{12}  &     &   \\
+u'_c  &  J_{21}  &   J_{22}  &     &   \\
+t'_w  &  J_{31}  &   J_{32}  &   J_{33}  &   J_{34}\\
+u'_w  &  J_{41}  &   J_{42}  &   J_{43} &   J_{44}
+\end{array}
+\]
+
+We will show that the absolute value of the determinant of this matrix is the same as the absolute value of the determinant of the following submatrix (which we will show is square), which much smaller than $J$ in the common case when a reversible jump move only modifies a subset of the random choices in a model:
+\[
+\left[
+\begin{array}{cc}
+J_{33}  &   J_{34}\\
+J_{43} &   J_{44}
+\end{array}
+\right]
+\]
+
+\begin{prop}
+The submatrix $[J_{11} \; J_{12}; J_{21} \; J_{22}]$ is square.
```

Comment:
Doesn't this follow from the fact that each "copy" takes one element of `t_c ∪ u_c` to one element of `t'_c ∪ u'_c`, and this is a one-to-one correspondence?  In other words, each "output of a copy" corresponds to exactly one "input to a copy."  Or am I misunderstanding?

Reply:
That's the intuition.

But syntactically, you are allowed to copy the same address to multiple destinations, so there the one-to-one correspondence is not an assumption. Instead, it derives from the assumption that the Jacobian determinant is nonzero (which is invoked in the proof of this proposition).

The argument could probably be simplified to:
(i) the assumption that the Jacobian determinant is nonzero implies that no two rows are the same
(ii) therefore the same address is not copied to multiple destination addresses, either explicitly or implicitly
(iii) then, t_c and u_c are elements that are copied (exactly once), and they are in one-to-one correspondence with elements of t_c' and u_c'
(iv) therefore, the submatrix is square

Reply:
Ah, got it.  Agreed, arguing directly that the correspondence is one-to-one makes things very clear.  Thanks!

Reply:
s/probably/definitely

Reply:
Done